src/DynamoCore/Engine/Profiling/NodeProfilingData.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using Dynamo.Core;
+using Dynamo.Graph.Nodes;
+using VMDataBridge;
+
+namespace Dynamo.Engine.Profiling
+{
+    class NodeProfilingData : NotificationObject, IDisposable
+    {
+        NodeModel node = null;
+
+        private DateTime? startTime = null;
+        private DateTime? endTime = null;
+
+        public NodeProfilingData(NodeModel node)
```

Comment:
these last three classes using the interfaces confuse me a bit - they are internal or private and they have public methods - how are they to be used or created?

please see:

https://stackoverflow.com/questions/16500593/what-does-a-public-constructor-on-an-internal-class-mean


also can we add summary /// to the constructors?

Reply:
if this was intentional until we want to open these APIs I am okay with that 👍 just pointing it out.